add missing header for fixed width integers

`vt` build (experimental alpine 3.13, clang-10, cmake 3.18.4) fails while compiling `example_3.cc`:
```
ninja: job failed: /usr/bin/ccache /usr/local/bin/mpicxx  -I/vt/lib/detector/src -Wall -pedantic -Wshadow -Wno-unknown-pragmas -Wsign-compare -ftemplate-backtrace-limit=100 -ftemplate-depth=900 -DCLI11_EXPERIMENTAL_OPTIONAL=0 -O3 -DNDEBUG -fcolor-diagnostics -MD -MT lib/detector/examples/CMakeFiles/example_3.dir/example_3.cc.o -MF lib/detector/examples/CMakeFiles/example_3.dir/example_3.cc.o.d -o lib/detector/examples/CMakeFiles/example_3.dir/example_3.cc.o -c /vt/lib/detector/examples/example_3.cc
/vt/lib/detector/examples/example_3.cc:93:24: error: unknown type name 'int32_t'
  TestConstructorNoIdx(int32_t, int32_t);
                       ^
```

Adding [standard header](https://en.cppreference.com/w/cpp/types/integer) fixes this:
```diff
--- a/examples/example_3.cc
+++ b/examples/example_3.cc
@@ -43,6 +43,7 @@
 */
 
 #include "detector.h"
+#include <cstdint>
 #include <cstdlib>
 
 namespace test {
```

--- a/examples/example_3.cc
+++ b/examples/example_3.cc
@@ -43,6 +43,7 @@
 */
 
 #include "detector.h"
+#include <cstdint>
 #include <cstdlib>
 
 namespace test {

